src/controllers/user-controller.ts

```diff
@@ -0,0 +1,116 @@
+import { Request, Response, NextFunction } from 'express';
+import { users, User } from '../db/users';
+import { v4 as uuidv4 } from 'uuid';
+import { userSchema } from '../models/user';
+
+export const getUsers = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void => {
+  const { loginSubstring, limit } = req.query;
+  const sortedUsers: User[] = users.sort((a: User, b: User) =>
+    a.login > b.login ? 1 : b.login > a.login ? -1 : 0,
+  );
+  const sortedAndFilteredUsers: User[] =
+    typeof loginSubstring === 'string'
+      ? sortedUsers.filter((user) =>
+          user.login
+            .replace(/ /g, '')
+            .toLowerCase()
+            .includes(loginSubstring.replace(/ /g, '').toLowerCase()),
+        )
+      : sortedUsers;
+  const responseData = !isNaN(Number(limit))
+    ? sortedAndFilteredUsers.slice(0, Number(limit))
+    : sortedAndFilteredUsers;
+  res.status(200).json(responseData.filter((user) => !user.isDeleted));
+  next();
+};
+
+export const getUser = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void => {
+  const selectedUser: User | undefined = users.find(
```

Comment:
```suggestion
  const user: User | undefined = users.find(
```

Applies to every other cases in this file

Reply:
done